addons/web/static/tests/views/kanban_view_tests.js

```diff
@@ -4307,7 +4307,22 @@ QUnit.module("Views", (hooks) => {
     });
 
     QUnit.test("prevent drag and drop of record if grouped by readonly", async (assert) => {
+        // Whether the kanban is grouped by state, foo or bar
+        // the user must not be able to drag and drop from one group to another,
+        // as state, foo or bar are made readonly one way or another.
+        // However, product_id must be draggable: by default, in the models, it's readonly,
+        // but a counter order is given in the view architecture: readonly="0".
```

Comment:
> Is it ok to overrule the `readonly: True` status in a view?

Yes it is allowed. Though I see this is not super common. Here is an example:

`odoobot_state` readonly in model:
https://github.com/odoo/odoo/blob/c9fa96604950f61c079627b88495130365d3b534/addons/mail_bot/models/res_users.py#L18

`odoobot_state` readonly overruled in view:
https://github.com/odoo/odoo/blob/c9fa96604950f61c079627b88495130365d3b534/addons/mail_bot/views/res_users_views.xml#L25

`odoobot_state` actually writable in the view (in debug mode):
![image](https://user-images.githubusercontent.com/5822488/191981827-6fb401e5-91bb-4c67-beb6-5010f68dbaf2.png)

Changing the value and saving works.

> What happens when we actually write on the field?

`readonly` is meant only for the views, in the web client. There is no protections client-side to prevent users to write on readonly fields what so ever. If you absolutely want to prevent user writing on a field, you must override `write` to prevent it.